lib/games.rb

```diff
@@ -206,38 +206,45 @@ def Game.open(host, port)
+                    if (Lich.display_lichid == true || Lich.display_uid == true)
+                      if XMLData.game =~ /^GS/
+                        if (Lich.display_lichid == true && Lich.display_uid == true)
+                          alt_string.sub!(/] \(\d+\)/) { "]" }
+                          alt_string.sub!(']') { " - #{Map.current.id}] (u#{(XMLData.room_id == 0 || XMLData.room_id > 4294967296) ? "nknown" : XMLData.room_id})" }
+                        elsif Lich.display_lichid == true
+                          alt_string.sub!(']') { " - #{Map.current.id}]" }
+                        elsif Lich.display_uid == true
+                          alt_string.sub!(/] \(\d+\)/) { "]" }
+                          alt_string.sub!(']') { "] (u#{(XMLData.room_id == 0 || XMLData.room_id > 4294967296) ? "nknown" : XMLData.room_id})" }
+                        end
                       end
-                    elsif XMLData.game =~ /^DR/
-                      @@room_number_after_ready = true
                     end
+                    @@room_number_after_ready = true
                   end
                   if @@room_number_after_ready && alt_string =~ /<prompt /
-                    room_number = ""
-                    room_number += "#{Map.current.id}" if Lich.display_lichid
-                    room_number += " - " if Lich.display_lichid && Lich.display_uid
-                    room_number += "#{XMLData.room_id}" if Lich.display_uid
-                    respond("Room Number: #{room_number}")
-                    room_exits = []
-                    Map.current.wayto.each_value do |value|
-                      if value.class != Proc
-                        # Don't include cardinals / up/down/out (usually just climb/go)
-                        room_exits << value if value !~ /^(?:o|d|u|n|ne|e|se|s|sw|w|nw|out|down|up|north|northeast|east|southeast|south|southwest|west|northwest)$/
+                    if XMLData.game =~ /^DR/
+                      room_number = ""
+                      room_number += "#{Map.current.id}" if Lich.display_lichid
+                      room_number += " - " if Lich.display_lichid && Lich.display_uid
+                      room_number += "#{XMLData.room_id}" if Lich.display_uid
+                      unless room_number.empty?
+                        alt_string = "Room Number: #{room_number}\r\n#{alt_string}"
+                        unless ['genie', 'frostbite'].include?($frontend)
+                          alt_string = "<streamWindow id='main' title='Story' subtitle=\" - [#{XMLData.room_title[2..-3]} - #{room_number}]\" location='center' target='drop'/>\r\n#{alt_string}"
+                          alt_string = "<streamWindow id='room' title='Room' subtitle=\" - [#{XMLData.room_title[2..-3]} - #{room_number}]\" location='center' target='drop' ifClosed='' resident='true'/>#{alt_string}"
+                        end
```

Comment:
This doesn't work for DR at all. To get our room number, the room title has to have been determined already. This isn't rewriting the room title to append the room number, it's just adding additional lines to the stream that aren't doing anything. It isn't possible at present to rewrite the room title window to append room number because room number cannot be determined before the whole room is parsed. It's why we have `Room Exits:` appended to the room description.

Reply:
Fair enough. Wrayth only makes sense.